### PR TITLE
Web - Fix selection jump on Chrome for Android

### PR DESCRIPTION
**Pull request by bleroux**
## Description

This PR fixes cursor jump on Chrome for Android when the user taps in a multiline `TextField`.

Using the following code sample:

<details><summary>Code sample</summary>

```dart

import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      title: 'Text Field Focus',
      home: MyCustomForm(),
    );
  }
}

// Define a custom Form widget.
class MyCustomForm extends StatelessWidget {
  const MyCustomForm({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Text Field Focus'),
      ),
      backgroundColor: Colors.amber,
      body: Padding(
        padding: const EdgeInsets.all(16.0),
        child: TextField(
          decoration: const InputDecoration(
            fillColor: Colors.white,
            filled: true
          ),
          autofocus: true,
          maxLines: 3,
          controller: TextEditingController(text: '1\n2\n3\n4\n'),
        ),
      ),// This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}

```
</details>

On a mobile browser, once the page is loaded, tap after the number 3: 

- Before this PR: the TextField content is automaticaly scrolled and the selection is set after number 1.

https://user-images.githubusercontent.com/840911/232051413-b913f890-6cb1-4c60-92d0-7a3bf74cc688.mov

## Implementation

A multiline `TextField` relies on an HTML `<textarea>` elements. When a tap occurs the selection should be updated from Flutter not by the HTML element itself. 
This PR prevents mouse events on Chrome for Android. Those events conflicts with Flutter selection changes.
Previously, mouse events were only prevented on desktop but they are also emitted on mobile, see https://bugs.chromium.org/p/chromium/issues/detail?id=119216#c11.

## Related Issue

Related to https://github.com/flutter/flutter/issues/124483 (partial fix because the issue is also reproducible on iOS/Safari).

## Tests

Adds 1 test.

### CODE REVIEW

File: lib/web_ui/test/engine/text_editing_test.dart [hunk 1]
```diff
@@ -2324,6 +2377,7 @@ Future<void> testMain() async {
       expect(textArea.selectionEnd, 2);
     });
 
+
```
Comment (kevmoo) [inline lib/web_ui/test/engine/text_editing_test.dart:2380]: extra new line

**Comment by kevmoo** [comment]
Thanks for doing the tweaks here!

**Comment by bleroux** [comment]
@mdebbar @kevmoo If you have some bandwidth, thank you to review this PR.

**Review by mdebbar — APPROVED** [review]
The change looks good to me, thanks for the contribution!

But I don't think it solves the linked issue. It only hides the symptom. Can this be done in `IOSTextEditingStrategy` too?

**Comment by bleroux** [comment]
@mdebbar Thank you for the review!
 
> But I don't think it solves the linked issue. It only hides the symptom. 

Do you have in mind another way to fix this? Because it might help to fix the iOS side.

>Can this be done in `IOSTextEditingStrategy` too?

This solution is not applicable on the iOS side because `IOSTextEditingStrategy` relies on DOM `click`event to tackle tap/longPress behaviors: 

https://github.com/flutter/engine/blob/40136ed00bc901b9f7c1752fe6315a69b329a429/lib/web_ui/lib/src/engine/text_editing/text_editing.dart#L1595-L1625

**Comment by mdebbar** [comment]
> Do you have in mind another way to fix this? Because it might help to fix the iOS side.

Well, that's a tricky issue that we've had for a while (see https://github.com/flutter/flutter/issues/99918 and https://github.com/flutter/flutter/issues/70841 for example). The only solution for it is to overhaul our text editing system (discussed in https://github.com/flutter/flutter/issues/120613).

**Pull request merged**